### ISSUE TITLE
dist train error in tutorial paddle 1.8

### ISSUE THREAD

**Issue by LiuHao-THU** [PaddlePaddle/Paddle#31678]
`
import paddle.fluid as fluid
# from nets import mlp
from paddle.fluid.incubate.fleet.parameter_server.distribute_transpiler import fleet
from paddle.fluid.incubate.fleet.base import role_maker
import numpy as np

def gen_data():
    return {"x": np.random.random(size=(128, 32)).astype('float32'),
            "y": np.random.randint(2, size=(128, 1)).astype('int64')}

def mlp(input_x, input_y, hid_dim=128, label_dim=2):
  fc_1 = fluid.layers.fc(input=input_x, size=hid_dim, act='tanh')
  fc_2 = fluid.layers.fc(input=fc_1, size=hid_dim, act='tanh')
  prediction = fluid.layers.fc(input=[fc_2], size=label_dim, act='softmax')
  cost = fluid.layers.cross_entropy(input=prediction, label=input_y)
  avg_cost = fluid.layers.mean(x=cost)
  return avg_cost
  
input_x = fluid.data(name="x", shape=[None, 32], dtype='float32')
input_y = fluid.data(name="y", shape=[None, 1], dtype='int64')

cost = mlp(input_x, input_y)
optimizer = fluid.optimizer.SGD(learning_rate=0.01)

role = role_maker.PaddleCloudRoleMaker()
fleet.init(role)
optimizer = fleet.distributed_optimizer(optimizer)
optimizer.minimize(cost)

if fleet.is_server():
  fleet.init_server()
  fleet.run_server()
elif fleet.is_worker():
  place = fluid.CPUPlace()
  exe = fluid.Executor(place)
  exe.run(fluid.default_startup_program())
  step = 1001
  for i in range(step):
    cost_val = exe.run(
        program=fluid.default_main_program(),
        feed=gen_data(),
        fetch_list=[cost.name])
    print("worker_index: %d, step%d cost = %f" %
         (fleet.worker_index(), i, cost_val[0]))
`

出现以下错误
`API is deprecated since 2.0.0 Please use FleetAPI instead.
WIKI: https://github.com/PaddlePaddle/Fleet/blob/develop/markdown_doc/transpiler

server not ready, wait 3 sec to retry...
not ready endpoints:['127.0.0.1:6170', '127.0.0.1:6171']
W0316 17:40:31.402160  2007 init.cc:216] Warning: PaddlePaddle catches a failure signal, it may not work properly
W0316 17:40:31.402179  2007 init.cc:218] You could check whether you killed PaddlePaddle thread/process accidentally or report the case to PaddlePaddle
W0316 17:40:31.402184  2007 init.cc:221] The detail failure signal is:

W0316 17:40:31.402189  2007 init.cc:224] *** Aborted at 1615916431 (unix time) try "date -d @1615916431" if you are using GNU date ***
W0316 17:40:31.403966  2007 init.cc:224] PC: @                0x0 (unknown)
W0316 17:40:31.420258  2007 init.cc:224] *** SIGSEGV (@0x0) received by PID 2007 (TID 0x7fc3967dc740) from PID 0; stack trace: ***
W0316 17:40:31.421954  2007 init.cc:224]     @     0x7fc3963b3980 (unknown)
W0316 17:40:31.435612  2007 init.cc:224]     @     0x7fc359121004 paddle::operators::SendOp::RunImpl()
W0316 17:40:31.446342  2007 init.cc:224]     @     0x7fc35aa0b261 paddle::framework::OperatorBase::Run()
W0316 17:40:31.458236  2007 init.cc:224]     @     0x7fc35749e200 paddle::framework::Executor::RunPartialPreparedContext()
W0316 17:40:31.464712  2007 init.cc:224]     @     0x7fc35749e5a3 paddle::framework::Executor::RunPreparedContext()
W0316 17:40:31.467535  2007 init.cc:224]     @     0x7fc3574a1d0b paddle::framework::Executor::Run()
W0316 17:40:31.475211  2007 init.cc:224]     @     0x7fc35717cfee _ZZN8pybind1112cpp_function10initializeIZN6paddle6pybindL22pybind11_init_core_avxERNS_6moduleEEUlRNS2_9framework8ExecutorERKNS6_11ProgramDescEPNS6_5ScopeEibbRKSt6vectorISsSaISsEEE111_vIS8_SB_SD_ibbSI_EINS_4nameENS_9is_methodENS_7siblingEEEEvOT_PFT0_DpT1_EDpRKT2_ENUlRNS_6detail13function_callEE1_4_FUNES10_
W0316 17:40:31.484406  2007 init.cc:224]     @     0x7fc3571df329 pybind11::cpp_function::dispatcher()
W0316 17:40:31.484771  2007 init.cc:224]     @     0x5602f33096e4 _PyMethodDef_RawFastCallKeywords
W0316 17:40:31.485119  2007 init.cc:224]     @     0x5602f3309801 _PyCFunction_FastCallKeywords
W0316 17:40:31.485469  2007 init.cc:224]     @     0x5602f33657ae _PyEval_EvalFrameDefault
W0316 17:40:31.485865  2007 init.cc:224]     @     0x5602f32a64f9 _PyEval_EvalCodeWithName
W0316 17:40:31.486235  2007 init.cc:224]     @     0x5602f3308a27 _PyFunction_FastCallKeywords
W0316 17:40:31.486656  2007 init.cc:224]     @     0x5602f33618fe _PyEval_EvalFrameDefault
W0316 17:40:31.487044  2007 init.cc:224]     @     0x5602f32a64f9 _PyEval_EvalCodeWithName
W0316 17:40:31.487422  2007 init.cc:224]     @     0x5602f3308a27 _PyFunction_FastCallKeywords
W0316 17:40:31.487839  2007 init.cc:224]     @     0x5602f33618fe _PyEval_EvalFrameDefault
W0316 17:40:31.488227  2007 init.cc:224]     @     0x5602f32a64f9 _PyEval_EvalCodeWithName
W0316 17:40:31.488586  2007 init.cc:224]     @     0x5602f3308a27 _PyFunction_FastCallKeywords
W0316 17:40:31.489003  2007 init.cc:224]     @     0x5602f33618fe _PyEval_EvalFrameDefault
W0316 17:40:31.489398  2007 init.cc:224]     @     0x5602f32a64f9 _PyEval_EvalCodeWithName
W0316 17:40:31.489809  2007 init.cc:224]     @     0x5602f32a73c4 PyEval_EvalCodeEx
W0316 17:40:31.490190  2007 init.cc:224]     @     0x5602f32a73ec PyEval_EvalCode
`

command
`python -m paddle.distributed.launch_ps --worker_num 2 --server_num 2 trainer.py`

link
https://www.paddlepaddle.org.cn/documentation/docs/zh/1.8/advanced_guide/distributed_training/fleet_api_howto_cn.html

**Comment by LiuHao-THU** [comment]
solved
